support-lib/wasm/djinni_wasm.cpp

```diff
@@ -188,11 +190,15 @@ EM_JS(void, djinni_register_name_in_ns, (const char* prefixedName, const char* n
 });
 
 void djinni_throw_native_exception(const std::exception& e) {
-    static auto throwCppException = em::val::module_property("djinni_throw_cpp_exception");
     if (const auto* jsEx = dynamic_cast<const JsException*>(&e)) {
-        throwCppException(jsEx->cause());
+        jsEx->cause().throw_();
     } else {
-        throwCppException(em::val(e.what()));
+        static std::exception_ptr exptr;
+        static auto ErrorClass = em::val::global("Error");
+        auto error = ErrorClass.new_(std::string("C++: ") + e.what());
+        exptr = std::current_exception();
```

Comment:
Technically this is the same exception as `e` but this gives us the ptr? We could also use `std::make_exception_ptr(e)` right? Looks good, just making sure I follow

Reply:
If I have an exception object that hasn't been thrown yet, I can use `make_exception_ptr` to get an `exception_ptr`.
If the exception is already thrown(like in our case), then `current_exception()` will return the `exception_ptr`.